### PR TITLE
Circuitpython port

### PR DESCRIPTION
**Pull request by TarjanPeter**
Some minor changes were needed to make it work on Circuitpython (I tested on an Adafruit Trinket M0). Mostly it was about replacing time.sleep_ms and time.sleep_us with time.sleep plus a few more odds and ends.

### CODE REVIEW

File: lcd/lcd_api.py [hunk 1]
```diff
@@ -1,195 +1,195 @@
...
+    def custom_char(self, location, charmap):
+        """Write a character to one of the 8 CGRAM locations, available
+        as chr(0) through chr(7).
+        """
+        location &= 0x7
+        self.hal_write_command(self.LCD_CGRAM | (location << 3))
+        time.sleep(0.00004)
+        for i in range(8):
+            self.hal_write_data(charmap[i])
+            time.sleep(0.00004)
```
[the remark this answers is not in the document]
Comment (TarjanPeter) [reply lcd/lcd_api.py]: I changed the EOLs (I think).  I don't understand what you mean by factoring the sleeping into the hal layer.
Sorry for the bumbling, I'm new to this.

**Comment by TarjanPeter** [comment]
Thanks for the clarification. Squashed and implemented.

File: lcd/circuitpython_i2c_lcd.py [hunk 2]
```diff
@@ -80,3 +80,8 @@ def hal_write_data(self, data):
         byte = (MASK_RS | (self.backlight << SHIFT_BACKLIGHT) | ((data & 0x0f) << SHIFT_DATA))
         self.i2c.writeto(self.i2c_addr, bytearray([byte | MASK_E]))
         self.i2c.writeto(self.i2c_addr, bytearray([byte]))
+		
+    def hal_sleep_us(self, data):
```
Comment (dhylands) [inline lcd/circuitpython_i2c_lcd.py]: I would rename data to be usecs. Otherwise this looks great.

File: lcd/lcd_api.py [hunk 3]
```diff
@@ -193,3 +193,7 @@ def hal_write_data(self, data):
         function.
         """
         raise NotImplementedError
+
+    def hal_sleep_us(self, data):
```
Comment (dhylands) [inline lcd/lcd_api.py]: Same thing here. Rename data to be usecs

Comment (TarjanPeter) [reply lcd/lcd_api.py]: Done and done.

File: lcd/circuitpython_i2c_lcd_test.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+from time import sleep, monotonic
+from circuitpython_i2c_lcd import I2cLcd
+
+# The PCF8574 has a jumper selectable address: 0x20 - 0x27
+DEFAULT_I2C_ADDR = 0x27
+
+def test_main():
+	"""Test function for verifying basic functionality."""
+	print("Running test_main")
+	i2c = busio.I2C(board.SCL, board.SDA)
```
Comment (dhylands) [inline lcd/circuitpython_i2c_lcd_test.py]: This file should use spaces rather than tabs for indentation (4 spaces is consistent with the remaining files)

**Pull request merged**